cirq-core/cirq/transformers/target_gatesets/compilation_target_gateset.py

```diff
@@ -170,10 +170,24 @@ def decompose_to_target_gateset(self, op: 'cirq.Operation', moment_idx: int) ->
         old_2q_gate_count = sum(1 for o in ops.flatten_to_ops(old_optree) if len(o.qubits) == 2)
         new_2q_gate_count = sum(1 for o in ops.flatten_to_ops(new_optree) if len(o.qubits) == 2)
         switch_to_new = (
-            any(op not in self for op in ops.flatten_to_ops(old_optree))
+            any(
+                protocols.num_qubits(op) == 2 and op not in self
+                for op in ops.flatten_to_ops(old_optree)
+            )
             or new_2q_gate_count < old_2q_gate_count
         )
-        return new_optree if switch_to_new else old_optree
+        if switch_to_new:
+            return new_optree
+        mapped_old_optree: List['cirq.OP_TREE'] = []
```

Comment:
This is different from the old behavior: if all old ops (1q or 2q) are in the target gateset and `new_2q_gate_count >= old_2q_gate_count`, this previously would have returned the old optree. Is this intended?

(If gateset behavior would prevent this case from occurring, a comment explaining this would be helpful)

Reply:
> this previously would have returned the old optree. 

The behavior is unchanged, i.e., if all old ops (1q or 2q) are in the target gateset and `new_2q_gate_count >= old_2q_gate_count`; it would have previously returned `old_optree` and it would still return `old_optree`. 

Notice that the `mapped_old_optree` will be same as `old_optree` if all operations are in `self` (the first if condition in the loop below).